### PR TITLE
Use index templates v2 in package 'base'

### PR DESCRIPTION
**Pull request by skh**
* Move the pre-built packages in `base` over to index templates v2
* introduces new directory `component-template`

**Do not merge before https://github.com/elastic/kibana/pull/64305 has been merged, as the current EPM implementation can't handle pre-built v2 packages**.

### CODE REVIEW

File: dev/packages/example/base-1.0.0/elasticsearch/component-template/events-mappings.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
+{
```
Comment (ruflin) [inline dev/packages/example/base-1.0.0/elasticsearch/component-template/events-mappings.json:1]: More general question we had around `ingest-pipeline` directory. Should it be `component_template` or `component-template`? What is our rule? Any preference?

Comment (skh) [reply dev/packages/example/base-1.0.0/elasticsearch/component-template/events-mappings.json:1]: I have no opinion here, I followed the example of `ingest-pipeline`.

(Kibana saved objects use hyphenation and not snake case, maybe that's how the initial format was derived even for assets that are not Kibana assets.)

**Comment by skh** [comment]
> My assumption is that in packages and EPM we ONLY support v2 templates. Is this correct?

Yes. For a smooth developer experience this week the code will be able to install pre-built v1 templates that are in the `base` package before this PR is merged, but other than  that everything will be v2.

File: dev/packages/example/base-1.0.0/elasticsearch/component-template/events-mappings.json (hunk 1 above)
Comment (ruflin) [reply dev/packages/example/base-1.0.0/elasticsearch/component-template/events-mappings.json:1]: lets keep the `-` for now. The part where I stumbled over `_` is in the ES API.

**Comment by ruflin** [comment]
@skh Conflict in the CHANGELOG. And because of my other PR, probably best rebase ...

File: dev/packages/example/base-1.0.0/elasticsearch/component-template/events-mappings.json (hunk 1 above)
Comment (ruflin) [inline dev/packages/example/base-1.0.0/elasticsearch/component-template/events-mappings.json:1]: After a rebase, this should become `base/1.0.0/..` as the path.

**Comment by skh** [comment]
Closed, see https://github.com/elastic/package-registry/pull/392

**Pull request closed without merging**